langchain4j-mistral-ai/src/test/java/dev/langchain4j/model/mistralai/MistralAiStreamingChatModelIT.java

```diff
@@ -0,0 +1,415 @@
+package dev.langchain4j.model.mistralai;
+
+import dev.langchain4j.data.message.AiMessage;
+import dev.langchain4j.data.message.UserMessage;
+import dev.langchain4j.model.StreamingResponseHandler;
+import dev.langchain4j.model.chat.StreamingChatLanguageModel;
+import dev.langchain4j.model.output.Response;
+import dev.langchain4j.model.output.TokenUsage;
+import org.junit.jupiter.api.Test;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeoutException;
+
+import static dev.langchain4j.data.message.UserMessage.userMessage;
+import static dev.langchain4j.model.output.FinishReason.LENGTH;
+import static dev.langchain4j.model.output.FinishReason.STOP;
+import static java.util.Arrays.asList;
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.assertj.core.api.Assertions.assertThat;
+
+class MistralAiStreamingChatModelIT {
+
+    StreamingChatLanguageModel model =  MistralAiStreamingChatModel.builder()
+            .apiKey(System.getenv("MISTRAL_AI_API_KEY"))
+            .temperature(0.1)
+            .build();
+
+    @Test
+    void should_stream_answer_and_return_token_usage_and_finish_reason_stop() throws ExecutionException, InterruptedException, TimeoutException {
+
+        CompletableFuture<String> futureAnswer = new CompletableFuture<>();
+        CompletableFuture<Response<AiMessage>> futureResponse = new CompletableFuture<>();
+
+        // given
+        UserMessage userMessage = userMessage("What is the capital of Peru?");
+
+        // when
+        model.generate(userMessage.text(), new StreamingResponseHandler<AiMessage>() {
```

Comment:
Some test `OpenAiStreamingChatModelIT` methods still use `StreamingResponseHandler` but I assume you're migrating them all to `TestStreamingResponseHandler`, right?

Reply:
yes, that's the plan. as well as making all tests model implementation agnostic, so that the same test logic applies to both blocking and streaming models.